src/daemon/scheduler.rs

```diff
@@ -169,230 +126,161 @@ fn make_cas_event_triggers(event_queue: Arc<MessageQueue>, ca_manager: Arc<CaMan
-) {
-    info!("Synchronize CA '{}' with its parent '{}'", ca, parent);
-    if let Err(e) = ca_manager.ca_sync_parent(&ca, &parent, actor).await {
-        let requeue_time = if test_mode_enabled() {
-            requeue_time_test()
-        } else {
-            requeue_time()
-        };
-
-        error!(
-            "Failed to synchronize CA '{}' with its parent '{}', error: {}",
-            ca, parent, e
-        );
-        event_queue.reschedule_sync_parent(ca, parent, requeue_time);
-    }
-}
-
-fn make_cas_republish(ca_server: Arc<CaManager>, event_queue: Arc<MessageQueue>) -> ScheduleHandle {
-    SkippingScheduler::run(
-        SCHEDULER_INTERVAL_SECONDS_REPUBLISH,
-        "CA certificate republish",
-        move || {
-            let rt = Runtime::new().unwrap();
-            rt.block_on(async {
-                debug!("Triggering background republication for all CAs, note this may be a no-op");
-                match ca_server.republish_all().await {
-                    Err(e) => error!("Background republishing of MFT and CRLs failed: {}", e),
-                    Ok(cas) => {
-                        for ca in cas {
-                            info!("Re-issued MFT and CRL for CA: {}", ca);
-                            event_queue.schedule_sync_repo(ca);
-                        }
-                    }
-                }
-            })
-        },
-    )
-}
-
-fn make_cas_objects_renew(ca_server: Arc<CaManager>, actor: Actor) -> ScheduleHandle {
-    SkippingScheduler::run(SCHEDULER_INTERVAL_SECONDS_ROA_RENEW, "CA ROA renewal", move || {
-        let rt = Runtime::new().unwrap();
-        rt.block_on(async {
-            debug!(
-                "Triggering background renewal for about to expire objects issued by all CAs, note this may be a no-op"
-            );
-            if let Err(e) = ca_server.renew_objects_all(&actor).await {
-                error!("Background re-issuing of about to expire objects failed: {}", e);
+    /// Queues tasks for background jobs when the server is started
+    async fn queue_start_tasks(&self) -> KrillResult<()> {
+        // If there are only a few CAs in this krill instance, then we
+        // will just want to re-sync them with their parents and repository
+        // on start up.
+        //
+        // If there are many, then will apply some random delays (jitter)
+        // to avoid a thundering herd. Note that the operator can always
+        // choose to run bulk operations manually if they know that they
+        // cannot wait.
+        let ca_list = self.ca_manager.ca_list(&self.system_actor)?;
+        let cas = ca_list.cas();
+
+        debug!("Adding tasks at start up");
+
+        let mut use_jitter = cas.len() >= SCHEDULER_USE_JITTER_CAS_THRESHOLD;
+
+        for summary in cas {
+            let ca = self.ca_manager.get_ca(summary.handle()).await?;
+
+            use_jitter = use_jitter || ca.nr_parents() >= self.config.ca_refresh_parents_batch_size;
```

Comment:
Should this be `let use_jitter = use_jitter...` ? As written I _think_ that a decision made for one CA whether to use jitter or not will affect the decision made for subsequent CAs in the loop but not previous CAs in the loop which seems a bit inconsistent.

Reply:
Good point - I changed the logic to only force this jitter for cas that have too many parents. And use the normal 'jitter' value for the repo sync.